rust/lance-index/src/vector/graph.rs

```diff
@@ -106,23 +113,12 @@ pub trait Graph<K: PrimInt = u32> {
     }
 
     /// Get the neighbors of a graph node, identifyied by the index.
-    fn neighbors(&self, key: K) -> Option<Box<dyn Iterator<Item = K> + '_>>;
+    fn neighbors(&self, key: u32) -> Option<Box<dyn Iterator<Item = &u32> + '_>>;
```

Comment:
I wonder if we could use u8 or u16 here. I don't image we go over `M=256` frequently

Reply:
we also get much better readahead with smaller types

Reply:
oh i see. makes sense